metadata/dashboard/dashboard_metadata.go

```diff
@@ -1650,61 +1571,70 @@ func (m *MetadataServer) GetTaskRuns(c *gin.Context) {
 		return
 	}
 
-	// todox: swap for api call
-	taskListCopy := make([]sc.TaskRunMetadata, len(taskRunStaticList))
-	_ = copy(taskListCopy, taskRunStaticList)
+	runs, err := m.client.Tasks.GetAllRuns()
+	if err != nil {
+		fetchError := m.GetRequestError(500, err, c, "GetTaskRuns - Failed to fetch task runs")
+		c.JSON(fetchError.StatusCode, fetchError.Error())
+		return
+	}
+
+	taskListResponse := make([]TaskRunResponse, 0)
+	for _, run := range runs {
+		task, err := m.client.Tasks.GetTaskByID(run.TaskId)
+		if err != nil {
+			fetchError := m.GetRequestError(500, err, c, "GetTaskRuns - Failed to fetch task")
```

Comment:
assuming the task id will be part of the error.

Reply:
yeah should be inside that error